api_app/analyzers_manager/file_analyzers/virushee.py

```diff
@@ -0,0 +1,84 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+import logging
+import time
+
+import requests
+
+from api_app.analyzers_manager.classes import FileAnalyzer
+from api_app.exceptions import AnalyzerRunException
+from tests.mock_utils import MockResponse, if_mock_connections, patch
+
+logger = logging.getLogger(__name__)
+
+
+class Virushee(FileAnalyzer):
+    """Analyze a file against Virushee API"""
+
+    def set_params(self, params):
+        self.session = requests.Session()
+        self.max_tries = 30
+        self.poll_distance = 10
+        self.request_url = "https://api.virushee.com/"
+
+    def run(self):
+        binary = self.read_file_bytes()
+        if not binary:
+            raise AnalyzerRunException("File is empty")
+        task_id = self.__upload_file(binary)
+        result = self.__poll_status(task_id)
+        return result
+
+    def __upload_file(self, binary) -> str:
+        name_to_send = self.filename if self.filename else self.md5
```

Comment:
you need to check the if the `hash` is already available before uploading it.

Reply:
This is the part where I'm facing problem. Can you refer to any existing example with such thing implemented.

Reply:
did you read the review comment?

> Did you read what was suggested in the related issue? #440 (comment)

> I mean, he completely described the flow to follow. What did you not understand?

Reply:
yes i did, in fact, I'm the one who told him to write that comment

Reply:
I've added a function `is_hash_avaliable` in the recent commit which returns true if a given hash is found but I'm unsure what to do if hash is found?
Should I just return the response??

Reply:
> if a given hash is found but I'm unsure what to do if hash is found?

here's the overall flow:
- given force_scan = False -> check hash -> return result if hash is found else upload file and then poll and return result
- given force_scan = True -> upload file -> poll and return result

Reply:
Okay, so in case of force_scan = False if hash is found I have to just return the result and complete the analysis. Got it